src/core/autosave/autosave.business.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  DatabaseSchemaVm,
+  createDefaultDatabaseSchemaVm,
+} from '@/core/providers';
+
+export interface CanvasSchema {
+  filename: string | undefined;
+  canvasSchema: DatabaseSchemaVm;
+}
+
+export const saveToLocal = (key: string, value: CanvasSchema) => {
+  try {
+    const deepCopyValue = JSON.parse(JSON.stringify(value));
+    const stringifiedDeepCopyValue = JSON.stringify(deepCopyValue);
+    localStorage.setItem(key, stringifiedDeepCopyValue);
+  } catch (error) {
+    console.error('Autosave error', error);
+    setTimeout(() => {
+      saveToLocal(key, value);
+    }, 30000);
```

Comment:
I think we should not call this in the catch, if something went wrong, well on next interval will try again